test/typing/pass/arithmetic_ops.py

```diff
@@ -0,0 +1,423 @@
+assert_type(FLOAT // BINARY, Binary)
+assert_type(FLOAT >= BINARY, Binary)
+assert_type(FLOAT > BINARY, Binary)
+assert_type(FLOAT <= BINARY, Binary)
+assert_type(FLOAT << BINARY, Binary)
+assert_type(FLOAT < BINARY, Binary)
+assert_type(FLOAT % BINARY, Binary)
+assert_type(FLOAT * BINARY, Binary)
+assert_type(FLOAT != BINARY, bool)
+assert_type(FLOAT | BINARY, Binary)
+assert_type(FLOAT**BINARY, Binary)
+assert_type(FLOAT >> BINARY, Binary)
+assert_type(FLOAT - BINARY, Binary)
+assert_type(FLOAT ^ BINARY, Binary)
+
+assert_type(BINARY + BOOL, Binary)
+assert_type(BINARY & BOOL, Binary)
+assert_type(BINARY / BOOL, Binary)
+assert_type(BINARY == BOOL, Binary)
+assert_type(BINARY // BOOL, Binary)
+assert_type(BINARY >= BOOL, Binary)
+assert_type(BINARY > BOOL, Binary)
+assert_type(BINARY <= BOOL, Binary)
+assert_type(BINARY << BOOL, Binary)
+assert_type(BINARY < BOOL, Binary)
+assert_type(BINARY % BOOL, Binary)
+assert_type(BINARY * BOOL, Binary)
+assert_type(BINARY != BOOL, Binary)
+assert_type(BINARY | BOOL, Binary)
+assert_type(BINARY**BOOL, Binary)
+assert_type(BINARY >> BOOL, Binary)
+assert_type(BINARY - BOOL, Binary)
+assert_type(BINARY ^ BOOL, Binary)
+
+assert_type(BOOL + BINARY, Binary)
+assert_type(BOOL & BINARY, Binary)
+assert_type(BOOL / BINARY, Binary)
+assert_type(BOOL == BINARY, bool)
+assert_type(BOOL // BINARY, Binary)
+assert_type(BOOL >= BINARY, Binary)
+assert_type(BOOL > BINARY, Binary)
+assert_type(BOOL <= BINARY, Binary)
+assert_type(BOOL << BINARY, Binary)
+assert_type(BOOL < BINARY, Binary)
+assert_type(BOOL % BINARY, Binary)
+assert_type(BOOL * BINARY, Binary)
+assert_type(BOOL != BINARY, bool)
+assert_type(BOOL | BINARY, Binary)
+assert_type(BOOL**BINARY, Binary)
+assert_type(BOOL >> BINARY, Binary)
+assert_type(BOOL - BINARY, Binary)
+assert_type(BOOL ^ BINARY, Binary)
+
+# Tensor operators whose types could be improved
+# This is the "diff" of the first and second sections.
+
+assert_type(BOOL // TENSOR, Any)
+assert_type(FLOAT // TENSOR, Any)
+assert_type(INT // TENSOR, Any)
+assert_type(TENSOR // BOOL, Any)
+assert_type(TENSOR // FLOAT, Any)
+assert_type(TENSOR // INT, Any)
+assert_type(TENSOR // TENSOR, Any)
+
+assert_type(BOOL**TENSOR, Any)
+assert_type(FLOAT**TENSOR, Any)
+assert_type(INT**TENSOR, Any)
+assert_type(TENSOR**BOOL, Any)
+assert_type(TENSOR**FLOAT, Any)
+assert_type(TENSOR**INT, Any)
+assert_type(TENSOR**TENSOR, Any)
+
+assert_type(BOOL - TENSOR, Any)
+assert_type(FLOAT - TENSOR, Any)
+assert_type(INT - TENSOR, Any)
+
+assert_type(BOOL / TENSOR, Any)
+assert_type(FLOAT / TENSOR, Any)
+assert_type(INT / TENSOR, Any)
+
+assert_type(BOOL % TENSOR, Any)
+assert_type(FLOAT % TENSOR, Any)
+assert_type(INT % TENSOR, Any)
+
+assert_type(BOOL << TENSOR, Any)
+assert_type(FLOAT << TENSOR, Any)
+assert_type(INT << TENSOR, Any)
+
+assert_type(BOOL >> TENSOR, Any)
+assert_type(FLOAT >> TENSOR, Any)
+assert_type(INT >> TENSOR, Any)
+
+assert_type(FLOAT & TENSOR, Tensor)  # type: ignore[operator]
```

Comment:
Feel like type ignores should be fixed at some point, right?

Reply:
@Skylion007 

Everything in that last section should be fixed! 😬 

Each line represents a quality error.  In this case, the deduced type is right, but the type checker wrongly believes that this operator doesn't exist.

Later, both the type and the existence of the operator are wrong. The demo code with the "ideal binary function" shows that it is possible to do...

:-D)